### PR TITLE
osd: fix logic in check_pg_upmaps

### PR DESCRIPTION
**Pull request by ljflores**
The logic was changed in check_pg_upmaps
in a Reef refactor, which results in recommendations made by the upmap balancer even when it says there are no optimizations.

Fixes: https://tracker.ceph.com/issues/63029

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Review by athanatos — APPROVED** [review]

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
jenkins test make check

File: src/osd/OSDMap.cc [hunk 1]
```diff
@@ -2157,7 +2157,7 @@ bool OSDMap::check_pg_upmaps(
                        << j->first << " " << j->second
                        << dendl;
         to_cancel->push_back(pg);
-      } else {
+      } else if (newmap != j->second) {
         //Josh--check partial no-op here.
```
Comment (JoshSalomon) [inline src/osd/OSDMap.cc:2161]: Please remove the "Josh" from the comment - it was an internal mark for me, but it is not needed anymore (I forgot to fix it)

Comment (ljflores) [reply src/osd/OSDMap.cc:2161]: Sure, will fix.

Comment (ljflores) [reply src/osd/OSDMap.cc]: Fixed.

**Comment by NitzanMordhai** [comment]
Rados review : https://tracker.ceph.com/projects/rados/wiki/MAIN#httpspulpitocephcomyuriw-2023-10-30_153436-rados-wip-yuri10-testing-2023-10-27-0804-distro-default-smithi

**Pull request merged**